### PR TITLE
Fix conda space check

### PR DESCRIPTION
**Pull request by huydhn**
Bash doesn't like float number, i.e. 3.5GB, so I switch to `df -k` to print the remaining KB as whole integer number. "God created the integers, all else is the work of man" :P

For example,

```
#!/bin/bash

if [ "3.5" -lt "4" ]; then
  echo "FOOBAR"
fi
```

prints the following warning `line 3: [: 3.5: integer expression expected`

This also fixes this miss https://github.com/pytorch/pytorch/actions/runs/3085913082/jobs/4989987822#step:1:32 in which the remaining space was measured in MB, so 585 (MB) was wrongly considered as larger than 3 (GB)

```
Print the available disk space for manual inspection
Filesystem       Size   Used  Avail Capacity iused      ifree %iused  Mounted on
/dev/disk5s2s1  100Gi   14Gi  585Mi    97%  501138    5991200    8%   /
devfs           219Ki  219Ki    0Bi   100%     756          0  100%   /dev
/dev/disk5s5    100Gi  2.0Gi  585Mi    78%       2    5991200    0%   /System/Volumes/VM
/dev/disk5s3    100Gi  296Mi  585Mi    34%     700    5991200    0%   /System/Volumes/Preboot
/dev/disk1s2    500Mi  6.0Mi  393Mi     2%       1    4028080    0%   /System/Volumes/xarts
/dev/disk1s1    500Mi   51Mi  393Mi    12%      53    4028080    0%   /System/Volumes/iSCPreboot
/dev/disk1s3    500Mi  224Ki  393Mi     1%      34    4028080    0%   /System/Volumes/Hardware
/dev/disk5s1    100Gi   82Gi  585Mi   100% 6[595](https://github.com/pytorch/pytorch/actions/runs/3085913082/jobs/4989987822#step:5:599)799    [599](https://github.com/pytorch/pytorch/actions/runs/3085913082/jobs/4989987822#step:5:603)1200   52%   /System/Volumes/Data
map auto_home     0Bi    0Bi    0Bi   100%       0          0  100%   /System/Volumes/Data/home
/dev/disk3s4    228Gi  276Ki  212Gi     1%      17 222[604](https://github.com/pytorch/pytorch/actions/runs/3085913082/jobs/4989987822#step:5:608)4560    0%   /private/tmp/tmp-mount-2RnJeF
There is 585Mi free space left in /, continue
```

### CODE REVIEW

File: .github/actions/setup-miniconda/action.yml [hunk 1]
```diff
@@ -73,21 +73,21 @@ runs:
           echo "Print the available disk space for manual inspection"
           df -h
 
-          MINIMUM_AVAILABLE_SPACE_IN_GB=3
+          # 3GB can still fail some tests, so set the minimum requirement space to 4GB
```
Comment (ZainRizvi) [inline .github/actions/setup-miniconda/action.yml:76]: nit: This comment might be more appropriate for the PR description.  A future maintainer would wonder "Yes we're using 4GB. Why did he mention 3GB?"

File: .github/actions/setup-miniconda/action.yml [hunk 2]
```diff
@@ -73,21 +73,21 @@ runs:
...
             AVAIL=$(echo $LINE | cut -f1 -d' ')
             MOUNT=$(echo $LINE | cut -f2 -d' ')
 
             if [ "$MOUNT" = "/" ]; then
-              AVAIL_IN_GB=${AVAIL::${#AVAIL}-2}
-
-              if [ "$AVAIL_IN_GB" -lt "$MINIMUM_AVAILABLE_SPACE_IN_GB" ]; then
-                echo "There is only $AVAIL free space left in $MOUNT, which is less than the minimum requirement of ${MINIMUM_AVAILABLE_SPACE_IN_GB}Gi. Please help create an issue to PyTorch Release Engineering via https://github.com/pytorch/test-infra/issues and provide the link to the workflow run."
+              if [ "$AVAIL" -lt "$MINIMUM_AVAILABLE_SPACE_IN_KB" ]; then
+                echo "There is only ${AVAIL}KB free space left in $MOUNT, which is less than the minimum requirement of ${MINIMUM_AVAILABLE_SPACE_IN_KB}KB. Please help create an issue to PyTorch Release Engineering via https://github.com/pytorch/test-infra/issues and provide the link to the workflow run."
```
Comment (ZainRizvi) [inline .github/actions/setup-miniconda/action.yml:87]: minor suggestion: `${MINIMUM_AVAILABLE_SPACE_IN_KB}KB.` might be hard to read

Instead, what if you set:

```
MINIMUM_AVAILABLE_SPACE_IN_GB=4
MINIMUM_AVAILABLE_SPACE_IN_KB=$(($MINIMUM_AVAILABLE_SPACE_IN_GB * 1024 * 1024))
```

and then had the comment say:
`minimum requirement of ${MINIMUM_AVAILABLE_SPACE_IN_GB}GB.`

**Review by ZainRizvi — APPROVED** [review]
A couple nits, otherwise looking good!

File: .github/actions/setup-miniconda/action.yml (hunk 1 above)
Comment (huydhn) [reply .github/actions/setup-miniconda/action.yml:76]: Ah ok, that makes sense :)

File: .github/actions/setup-miniconda/action.yml (hunk 2 above)
Comment (huydhn) [reply .github/actions/setup-miniconda/action.yml:87]: Sure, I'll update this accordingly

**Pull request merged**